src/views/Login/Login.tsx

```diff
@@ -21,101 +23,121 @@ const autocompleteSx = {
 );
 
 const Login: React.FC = () => {
   const { flushFn, userLogin } = useAccount();
   const [accounts, setAccounts] = useLocalStorage<Array<string>>("accounts", []); // Stores user accounts in localStorage under "accounts" key
   const [existingUser, setExistingUser] = useState<"existing" | "new">("new");
+  const [existingUserType, setExistingUserType] = useState<"address" | "secretPhrase">("address");
   const [userInputAccount, setUserInputAccount] = useState<string>("");
-  const [isValidAddressState, setIsValidAddressState] = useState<boolean>(false);
+  const [isValidInputState, setIsValidInputState] = useState<boolean>(false);
   const [userRememberState, setUserRememberState] = useState<boolean>(false);
   const { enqueueSnackbar } = useSnackbar();
 
-  const handleAddressInput = useCallback((newValue: string | null) => {
-    if (newValue === null) {
-      return;
-    }
-
-    if (!isValidAddress(newValue)) {
-      setIsValidAddressState(false);
-    } else {
-      setIsValidAddressState(true);
-    }
-
-    setUserInputAccount(newValue);
-
-    /*
-     * TODO: Add formatting to address entry
-     * - Auto uppercase
-     * - 'JUP-' prefixing and hyphens at appropriate positions
-     *
-     * if (newValue.length > 2) {
-     *   newValue += "-";
-     * }
-     */
-  }, []);
+  const handleAddressOrSecretInput = useCallback(
+    (newValue: string | null) => {
+      if (newValue === null) {
+        return;
+      }
+      setUserInputAccount(newValue);
+
+      if (existingUserType === "address") {
+        if (!isValidAddress(newValue)) {
+          setIsValidInputState(false);
+          return;
+        } else {
+          setIsValidInputState(true);
+        }
+      } else if (existingUserType === "secretPhrase") {
+        if (!isValidSecret(newValue)) {
+          enqueueSnackbar(messageText.validation.secretLengthWarning, { variant: "warning" });
+          setIsValidInputState(true); // we still allow the user to login after showing them a warning
+          return;
+        } else {
+          setIsValidInputState(true);
+        }
+      }
+    },
+    [enqueueSnackbar, existingUserType]
+  );
 
   const handleExistingUserChoiceFn = useCallback(() => {
     setExistingUser((prev) => (prev === "new" ? "existing" : "new"));
   }, []);
 
+  const handleExistingUserTypeChoiceFn = useCallback(() => {
+    setExistingUserType((prev) => (prev === "secretPhrase" ? "address" : "secretPhrase"));
+  }, []);
+
   /*
    * If the user has selected the "remember me" checkbox this will save their input entry in localStorage
    * Otherwise just validates their address and proceeds them to the dashboard
    */
   const handleLogin = useCallback(
     (e) => {
-      // Address is NOT valid
-      if (!isValidAddress(userInputAccount)) {
-        e.preventDefault(); // Prevents navigation to Dashboard
-        setIsValidAddressState(false);
-        return;
+      let accountRs = "";
+      if (existingUserType === "address") {
```

Comment:
this extra layer of if statement makes this a bit more ugly. the only suggestion i can see to clean it up a bit is to make 2 callbacks, and with your if statement, call the proper one to handle the login of the given type.

Reply:
Refactored, it does help readability a bit, let me know if the way I did it is what you had in mind.

Reply:
that's essentially what i was thinking. I might have used a switch statement instead of the if else if, but it's not a big deal either way. thanks!